packages/http/src/client.ts

```diff
@@ -30,16 +30,25 @@ export function createClientFromOperations(resources: IHttpOperation[], defaultC
 
   const client: PrismHttp = {
     request(url, input, config) {
-      const parsedUrl = parseUrl(url);
-
-      if (!parsedUrl.pathname) throw new Error('Path name must always be specified');
+      if (!url) throw new Error('Path name must always be specified');
 
       const mergedConf: IClientConfig = defaults(config, defaultConfig);
+      const baseUrl = pipe(
+        O.tryCatch(() => new URL(url)),
+        O.fold(
+          () => mergedConf.baseUrl,
+          url => url.origin
+        )
+      );
+      const parsedUrl = pipe(
+        O.tryCatch(() => new URL(url)),
+        O.getOrElseW(() => new URL(url, 'https://stoplight.io'))
+      );
 
       const httpUrl: IHttpUrl = {
-        baseUrl: parsedUrl.host ? `${parsedUrl.protocol}//${parsedUrl.host}` : mergedConf.baseUrl,
+        baseUrl,
         path: parsedUrl.pathname,
-        query: parseQueryString(parsedUrl.query || '') as IHttpNameValues,
+        query: parseQueryString(parsedUrl.search.substring(1)) as IHttpNameValues,
```

Comment:
is `search` guaranteed not to be undefined?

Reply:
Yes, I found that it is always string even though there are no query string in URL. 
but I think it is better to make code more defensive 😄